launch/launch/actions/include_launch_description.py

```diff
@@ -106,19 +106,30 @@ def visit(self, context: LaunchContext) -> List[LaunchDescriptionEntity]:
 
         # Do best effort checking to see if non-optional, non-default declared arguments
         # are being satisfied.
-        argument_names = [
-            perform_substitutions(context, normalize_to_list_of_substitutions(arg_name))
-            for arg_name, arg_value in self.launch_arguments
-        ]
-        declared_launch_arguments = launch_description.get_launch_arguments()
+        declared_launch_arguments = launch_description.get_launch_arguments(
+            include_launch_description_parent=self
+        )
         for argument in declared_launch_arguments:
             if argument._conditionally_included or argument.default_value is not None:
                 continue
+            argument_names: List[str] = []
+            if argument._include_launch_description_parent is not None:
+                argument_names = [
+                    perform_substitutions(context, normalize_to_list_of_substitutions(arg_name))
+                    for arg_name, arg_value
+                    in argument._include_launch_description_parent.launch_arguments
```

Comment:
The problem that I see here is the following:
An argument declared in a nested `IncludeLaunchDescription`, it's automatically an argument of the parent launch_description. So I expect that if the included launch description has a "grandparent", which was included passing the remaining `launch_arguments` that weren't solved by the "parent", it should work. But it isn't working in that way, because it is not checking if `argument._include_launch_description_parent` has a parent and recursively loading its arguments. Clearer example [here](https://gist.github.com/ivanpauno/c3840eec2cea59c8a1e32b96b1f26f0e).

Reply:
I see your point... this is also connected to whether or not we want to require explicitly passing an argument from the parent launch file into the included one or not.

We may not have time to address all of these cases before the distro freeze. In light of that, @ivanpauno do you want to just use #249? Or do you think we should go forward with this pr even though it doesn't support the case you put into your example?

Reply:
Between this as-is and #249, I prefer the second one.
#249 solves #248, and it also have a consistent behavior in the example I give above ([here again](https://gist.github.com/ivanpauno/c3840eec2cea59c8a1e32b96b1f26f0e)). A call to `ros2 launch --show-args` only shows the arguments you can really set (in the example above, `other2`). So calling `ros2 launch package_name test.launch.py problematic_argument:='asd'` will fail when including `other.launch.py` from `test.launch.py`, which is at least consistent with the idea of not detecting nested arguments in `IncludeLaunchDescription` actions.

If the final goal is to automatically recognize the nested arguments in others `IncludeLaunchDescription` actions, we will not break existing code when introducing that (duplicated declarations aren't a problem).

I would like to hear another opinion before continuing with #249, maybe @hidmic?

Reply:
Hmm, on one hand we have this PR, that allows implicit propagation of arguments, which wasn't supported on ROS1 but IMHO is a really cool feature i.e. it reduces the maintenance burden for large hierarchies of launch files, specially if including third party ones. However, IIUC it's currently limited to a depth of 1, which is at very least unexpected. So I'd be inclined to go with #249, being as limited as ROS 1, and hopefully deliver this PR's functionality in an upcoming patch release.